✨ Feat: svgr 이용해 svg 파일 컴포넌트화 환경 설정

2gyoung opened the pull request:
## 1. svgr 환경 설정
![image](https://github.com/user-attachments/assets/505db5b3-d3bb-416a-a47e-5fc348d0f361)

- svg 파일을 컴포넌트처럼 사용하기 위해 svgr을 이용해 환경 설정을 진행했습니다.

## 2. svg 파일명 통일
- svg 파일의 파일명을 kebab-case로 통일했습니다. 
- 파일 내에 svg를 컴포넌트로 import할 경우에는 `PascalCase`로 import해야 합니다.
- 위 예시에 대한 코드입니다. 

![image](https://github.com/user-attachments/assets/13110886-f0b6-4cf9-bdc7-a88d6838e31a)
![image](https://github.com/user-attachments/assets/ed02852b-6f6d-4e1b-a45f-3c00e67e9cb5)

## 3. declare 파일 이동 
- '@types' 폴더를 신설하여 declare 파일을 이동시켰습니다. 
- 현재 이동된 파일은 `image.d.ts`와 `svgr.d.ts`입니다.

2gyoung committed:
✨Feat : svg 시범 추가

- 8개 svg에 대해 시범적 추가
- unexpected token 에러 발생함

2gyoung committed:
🚚Rename: svg 파일명 kebab-case로 변경

2gyoung committed:
✨ Feat: svg 파일 컴포넌트화 설정

hyeonseok98 commented on tsconfig.json:
```diff
@@ -23,6 +23,6 @@
     }
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts", "src"],
+  "include": ["src/global.d.ts", "next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts", "src"],
```
경로가 root에 위치해있어서 src/는 제거해도 될 것 같아요

2gyoung replied:
말씀 주신 부분들 적용했습니다. images.d.ts 파일을 'node_modules/@types/' 하위로 이동하고 관련 경로도 수정했으나, node_modules 폴더가 .gitignore에 등록되어 있어서 pull했을 때 해당 파일이 존재하지 않을 수 있어 우려됩니다. pull 이후 images.d.ts 파일이 @types 폴더 하위에 존재하지 않는다면 수동 복사 부탁드립니다 ^_^;;

hyeonseok98 replied:
스크린샷과 활용 코드 감사합니다!! 덕분에 이해가 잘 됐어요

@types 폴더는 @node_module에 있는 거랑 별개로, 사용자 지정 d.ts 파일을 한 곳에 몰아두면 어떨까 해서 제안드려봤어요. 현재 images.d.ts와 global.d.ts가 있어서 모아 놓는 곳이 있으면 좋을 것 같았거든요. 때문에 /@types/images.d.ts 이런 경로로 따로 만들면 어떨까 합니다

+ 추가로 global.d.ts가 전역적으로 사용되는 global이라는 것보다는 svgr에 한정되어 있어서 svgr.d.ts와 같은 네이밍을 해도 괜찮을 것 같아서 의견 남겨봐요! 고생하셨습니다

hyeonseok98 reviewed (commented):
작업 고생 많으셨습니다~!!

1. svgr 컴포넌트가 잘 작동하는지 props까지 넣은 테스트 스크린샷이 있으면 좋을 것 같아요. 어떻게 사용할 수 있는지 방법을 공유해주시면 감사하겠습니다.
2. images.d.ts를 사용한다면 svg도 추가되면 좋을 것 같아요!
3. 000.d.ts는 root에 @types 내 정의해주셔도 좋을 것 같아요. d.ts 파일은 해당 폴더 내 위치해서 많이들 사용된다고 하네요.(@types 폴더라고 검색해보면 나오는 것 같아요)

2gyoung committed:
🔨 Chore: svg 관련 d.ts 파일 참조 경로 수정

- global.d.ts 참조 경로 수정 ('src/' 제거)
- images.d.ts 참조 경로 'node_modules/@types/' 아래로 이동

2gyoung committed:
✨ Feat: svgr 예시 코드 작성 및 종속성 업데이트

- main 페이지에 svg 컴포넌트 적용한 예시 코드 작성
- svgr 관련 라이브러리와 종속성 버전 업데이트

2gyoung committed:
🚚 Rename: node_modules/@types 폴더 하위로 파일 이동

2gyoung committed:
🚚 Rename: @types 폴더 신설 후 파일 이동

- declare 파일 이동
